exercises/practice/complex-numbers/.meta/test_template.ecr

```diff
@@ -0,0 +1,77 @@
+require "spec"
+require "../src/*"
+
+<% property_map = {
+  ".real" => ".real",
+  ".imaginary" => ".imaginary",
+  ".abs" => ".abs",
+  ".conjugate" => ".conjugate",
+  ".exp" => ".exp",
+  ".add" => "+",
+  ".sub" => "-",
+  ".mul" => "*",
+  ".div" => "/"
+} %>
+
+<%
+complex_args = ->(json : JSON::Any) {
+  json
+    .as_a
+    .map(&.to_s)
+    .map(&.gsub("e", "Math::E"))
+    .map(&.gsub("pi", "Math::PI"))
+    .map(&.gsub("ln", "Math.log"))
+    .join(",")
+}
+%>
+
+describe <%= to_capitalized(@json["exercise"].to_s).inspect %> do
+    <% flattened_cases = [] of JSON::Any %>
+
+    <% @json["cases"].as_a.each do |cases| %>
+        <% if cases["cases"]? %>
+            <% cases["cases"].as_a.each do |cases| %>
+                <% if cases["cases"]? %>
+                        <% flattened_cases.concat(cases["cases"].as_a) %>
+                <% else -%>
+                        <% flattened_cases.concat([cases]) %>
+                <% end %>
+            <% end %>
+        <% else -%>
+            <% flattened_cases.concat([cases]) %>
+        <% end %>
+    <% end %>
```

Comment:
The structure of the canonical data is not friendly for automation. I added this to flatten the structure so I wouldn't need to duplicate the logic for creating each test case (which is already very complex) once for every level of nesting.

Reply:
I think it is fine, I have seen the problem spec repo for this exercise and it is structured in a way that makes it a bit complicated.

Reply:
If you like you could add a method here if you feel like the template becomes a bit complicated: https://github.com/exercism/crystal/blob/main/bin/generator_plugins.cr

Reply:
I'll do that. I realize now that this uses ECR in a way that's unnecessary and it could instead just use Crystal.